### PR TITLE
chore(suite): refactor staking

### LINKED ISSUE #15398: Unused input props drilled to DOM
1. Open console
2. Go to send form

Caused by https://github.com/trezor/trezor-suite/commit/f7cb5caeff7a028655ac58f52a46012fc4a23008

Screenshot 2024-11-14 at 16 40 31
Screenshot 2024-11-14 at 16 30 51
Screenshot 2024-11-14 at 16 30 59
Screenshot 2024-11-14 at 16 31 10
Screenshot 2024-11-14 at 16 31 24
Screenshot 2024-11-14 at 16 31 31
Screenshot 2024-11-14 at 16 31 38

### PR DESCRIPTION
**Pull request by seibei-iguchi**
- refactored staking dashboard and claim modal
- added flex to frame props
- added paddingType to IconCircle component
- added iconName to InfoRow component
- removed hasBottomPadding from form components
- fixed prop passing in form components

Resolves https://github.com/trezor/trezor-suite/issues/15398

### Before
Screenshot 2024-11-14 at 14 11 24
Screenshot 2024-11-14 at 14 11 38

### After
Screenshot 2024-11-14 at 13 57 35
Screenshot 2024-11-14 at 13 58 18

### CODE REVIEW

**Review by tomasklim — CHANGES_REQUESTED** [review]
Claiming is usually not available, so most of the users will see it like this 
Screenshot 2024-11-15 at 8 51 31

This banner looks weird in the bottom
At the top or better at the modal header as description, it would maybe look better?
Screenshot 2024-11-15 at 8 47 40

Probably you removed it in the previous refactoring but the fee component had some sort of state that the previous gas limit and gas price was visible before the new one was composed. That was there to avoid this jumping

https://github.com/user-attachments/assets/fac9f013-b4d1-4e37-aeee-66d8fe7f9bd5

**Comment by seibei-iguchi** [comment]
> Claiming is usually not available, so most of the users will see it like this 

Ok, I didn't know that, I thought there is always either claiming available or pending:) I'll fix that.

File: packages/components/src/components/form/Select/Select.tsx [hunk 1]
```diff
@@ -280,7 +280,6 @@ export type SelectProps = KeyPressScrollProps &
         isClean?: boolean;
         label?: ReactNode;
         size?: InputSize;
-        hasBottomPadding?: boolean;
```
Comment (jvaclavik) [inline packages/components/src/components/form/Select/Select.tsx:283]: Are you sure about this change? We used both values.

**Comment by tomasklim** [comment]
@trezor/qa this PR should be just staking UI/UX changes, but please test it properly 🙏

**Review by tomasklim — APPROVED** [review]
LGTM

Would be nice to find a solution for this claiming banner as it is huge but I do not want to block merging to avoid conflicts
Screenshot 2024-11-15 at 17 54 00

**Review by jvaclavik — APPROVED** [review]

**Comment by bosomt** [comment]
**QA OK**

- stake
- unstake
- claim
- bump
- dark mode

image

image

image

**Info:**
 - Suite version: web 24.12.0 (182d6b4026aad83ca7544ee77419624659308e35)
 - Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/131.0.0.0 Safari/537.36
 - OS: MacIntel
 - Screen: 1512x982
 - Device: Trezor T3T1 2.8.3 regular (revision 7f373ae71eca855dd41b4a0fdcc7cadaa13a8281)
 - Transport: WebUsbTransport

**Pull request merged**